src/main/java/org/terasology/fluid/system/FluidClientSystem.java

```diff
@@ -122,4 +130,45 @@ private void setFluidContainerIcon(EntityRef container, FluidContainerItemCompon
             }
         }
     }
+
+    /**
+     * Used to draw the Filling bar over the fluid container item in the cell.
+     *
+     * @param event  An event sent after the inventory cell has been rendered.
+     * @param entity The entity sending the request.
+     * @param fluidContainer FluidContainerItemComponent of the item.
+     */
+    @ReceiveEvent
+    public void drawFillingBarForFluidContainerItem(InventoryCellRendered event, EntityRef entity,
+                                                    FluidContainerItemComponent fluidContainer) {
+        Canvas canvas = event.getCanvas();
+
+        Vector2i size = canvas.size();
+
+        int minX = (int) (size.x * 0.8f);
+        int maxX = (int) (size.x * 0.9f);
+
+        int minY = (int) (size.y * 0.1f);
+        int maxY = (int) (size.y * 0.9f);
+
+        float fillingPercentage = 1f * fluidContainer.volume / fluidContainer.maxVolume;
+
+        if (fillingPercentage != 1f) {
+            ResourceUrn backgroundTexture = TextureUtil.getTextureUriForColor(Color.WHITE);
+            final Color terasologyColor = getTerasologyColorForFilledAmount(fillingPercentage);
+            ResourceUrn barTexture = TextureUtil.getTextureUriForColor(terasologyColor);
+
+            canvas.drawTexture(Assets.get(backgroundTexture, Texture.class).get(), Rect2i.createFromMinAndMax(minX,
+                    minY, maxX, maxY));
+            int fillingBarHeight = (int) (fillingPercentage * (maxY - minY - 1));
+            int fillingBarLength = maxX - minX - 1;
+            canvas.drawTexture(Assets.get(barTexture, Texture.class).get(), Rect2i.createFromMinAndSize(minX + 1,
+                    maxY - fillingBarHeight - 1, fillingBarLength, fillingBarHeight ));
+        }
+    }
+
+    private Color getTerasologyColorForFilledAmount(float fillingPercentage) {
+        final java.awt.Color awtColor = java.awt.Color.getHSBColor(0.33f * fillingPercentage, 1f, 0.8f);
```

Comment:
It would be nice to add a static method to the Terasology NUI Color class `Color::getHSBColor`. I cannot imagine this is the only place where we are doing something like this... 🤔

Reply:
So just want me to add something like this in Color.java right?:
>  public Color getHSBColor(float h, float s, float b) {
        final java.awt.Color awtColor = java.awt.Color.getHSBColor(h, s, b);
        return new Color(awtColor.getRed(), awtColor.getGreen(), awtColor.getBlue());
    }